DDECal/TECConstraint.cc

```diff
@@ -53,8 +54,21 @@ void ApproximateTECConstraint::initializeChild()
 
 void TECConstraintBase::applyReferenceAntenna(std::vector<std::vector<dcomplex> >& solutions) const
 {
-  // TODO chose this more cleverly?
+  // Choose reference antenna that has at least 20% channels unflagged
   size_t refAntenna = 0;
+  for(; refAntenna!=_nAntennas; ++refAntenna)
+  {
+    size_t nUnFlaggedChannels=0;
+    // Only check flagged state for first direction
+    for (size_t ch=0; ch!=_nChannelBlocks; ++ch)
+    {
+      if(isfinite(solutions[0][refAntenna*_nDirections]))
```

Comment:
Something is wrong here; `ch` is not used.

Reply:
Will change to `ch`